packages/wallets/torus/src/adapter.ts

```diff
@@ -1,109 +1,98 @@
 import {
+    BaseMessageSignerWalletAdapter,
     BaseSignerWalletAdapter,
+    EventEmitter,
+    pollUntilReady,
+    WalletAccountError,
     WalletConnectionError,
+    WalletDisconnectedError,
     WalletDisconnectionError,
     WalletError,
-    WalletKeypairError,
     WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
     WalletSignTransactionError,
-    WalletWindowBlockedError,
     WalletWindowClosedError,
 } from '@solana/wallet-adapter-base';
-import { Keypair, PublicKey, Transaction } from '@solana/web3.js';
-import OpenLogin, { OPENLOGIN_NETWORK, OpenLoginOptions } from '@toruslabs/openlogin';
-import { getED25519Key } from '@toruslabs/openlogin-ed25519';
+import { PublicKey, Transaction, Message, Cluster } from '@solana/web3.js';
+import Torus, { TorusParams } from '@toruslabs/solana-embed';
 
 export interface TorusWalletAdapterConfig {
-    options: Partial<OpenLoginOptions> & Omit<OpenLoginOptions, 'network'>;
+    torusParams?: TorusParams;
+    // network?: WalletAdapterNetwork;
+    pollInterval?: number;
+    pollCount?: number;
 }
+interface TorusWindow extends Window {
+    torus: Torus | null;
+}
+
+declare const window: TorusWindow;
 
 export class TorusWalletAdapter extends BaseSignerWalletAdapter {
-    private _options: OpenLoginOptions;
     private _connecting: boolean;
-    private _openLogin: OpenLogin | null;
-    private _keypair: Keypair | null;
+    private _torus: Torus | null;
+    private _publicKey: PublicKey | null;
+    private _config: TorusParams;
 
     constructor(config: TorusWalletAdapterConfig) {
         super();
-        this._options = { uxMode: 'popup', network: OPENLOGIN_NETWORK.MAINNET, ...config.options };
         this._connecting = false;
-        this._openLogin = null;
-        this._keypair = null;
+        this._torus = null;
+        this._publicKey = null;
+        this._config = config.torusParams || {};
+        // if (!this.ready) pollUntilReady(this, config.pollInterval || 1000, config.pollCount || 3);
+        // this.ready = true;
```

Comment:
Likewise, if polling isn't needed, it can be removed, along with the config for it.